local_planner/src/nodes/common.cpp

```diff
@@ -115,6 +115,22 @@ void createPoseMsg(Eigen::Vector3f& out_waypt, Eigen::Quaternionf& out_q,
           Eigen::AngleAxisf(yaw, Eigen::Vector3f::UnitZ());
 }
 
+float getYawFromQuaternion(const Eigen::Quaternionf q){
+	tf::Quaternion tf_q(q.x(), q.y(), q.z(), q.w());
+	tf::Matrix3x3 tf_m(tf_q);
+	double roll, pitch, yaw;
+	tf_m.getRPY(roll, pitch, yaw);
+	return static_cast<float>(yaw);
+}
+
+float getPitchFromQuaternion(const Eigen::Quaternionf q){
+	tf::Quaternion tf_q(q.x(), q.y(), q.z(), q.w());
```

Comment:
Again, from wiki: (https://en.wikipedia.org/wiki/Conversion_between_quaternions_and_Euler_angles)
(Maybe we'll have to verify the frame and convention first, but I'd prefer doing this in two simple lines as opposed to calling the tf libraries)
```
	double sinp = +2.0 * (q.w() * q.y() - q.z() * q.x());
	if (fabs(sinp) >= 1)
		pitch = copysign(M_PI / 2, sinp); // use 90 degrees if out of range
	else
		pitch = asin(sinp);
```